src/sonic-py-common/sonic_py_common/interface.py

```diff
@@ -88,5 +105,30 @@ def get_port_table_name(interface_name):
         return "VLAN_INTERFACE"
     elif interface_name.startswith(loopback_prefix()):
         return "LOOPBACK_INTERFACE"
+    elif VLAN_SUB_INTERFACE_SEPARATOR in interface_name:
+        if interface_name.startswith(physical_subinterface_prefix()) or interface_name.startswith(portchannel_subinterface_prefix()):
+            return "VLAN_SUB_INTERFACE"
     else:
         return ""
+
+def intf_get_longname(intf):
+    if intf is None:
+       return None
+
+    if VLAN_SUB_INTERFACE_SEPARATOR in intf:
```

Comment:
Do we need to check Eth or Po prefix as well instead of blindly checking for "." in the intf string and returning long interface name?

Reply:
In certain case we need to fetch full name for the given subinterface parent interface which could be in short name format. Just checking for "." will not suffice here.